File: 0378/378.go
func kthSmallestElementInASortedMat(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := kthSmallestElementInASortedMat(root.Left)
	r := kthSmallestElementInASortedMat(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 0378/378.c
int kthSmallestElementInASortedMat(struct TreeNode* root) {
    if (!root) return 0;
    int l = kthSmallestElementInASortedMat(root->left);
    int r = kthSmallestElementInASortedMat(root->right);
    return 1 + (l > r ? l : r);
}